processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1051,24 +1050,19 @@ public MethodReference getFinalizerMethod() {
     }
 
     public List<Parameter> getSourceParametersExcludingPrimitives() {
-        List<Parameter> sourceParameters = new ArrayList<>();
-        for ( Parameter sourceParam : getSourceParameters() ) {
-            if ( !sourceParam.getType().isPrimitive() ) {
-                sourceParameters.add( sourceParam );
-            }
-        }
+        Predicate<Parameter> excludePrimitivePredicate = parameter -> !parameter.getType().isPrimitive();
```

Comment:
I am not sure how I feel about adding this predicate like this here and using `filterSourceParameters`. Perhaps if we remove the `filterSourceParameters` and do `return getSourceParameters().filter( param -> !param.getType().isPrimitive ).collect( Collectors.toList() );` instead. It would be more readable.

Same comments is valid for the other places where we pass a `Predicate`

Reply:
@filiphr You are right. I have done it